e2e_testing/README.md

```diff
@@ -1 +1,87 @@
-# e2e_testing
+# E2E Testing
+
+The E2E Tests bring the web front end and backing services under test and emulate a user interacting with the browser.
+
+They are designed to run in two modes:
+
+- _Pre-release_ - During CI the tests run against locally spun up services including an ephemeral database that includes the Django web service schema migrations plus any test fixture data for the tests to make assertions against.
+
+- _Post-release_ - A subset of the tests can be run against a live environment to sanity check deployments and confirm services are running as expected. These are intended to be non destructive with regard to the database and so should not make changes to data or be dependent on any pre-existing data (see Tag Annotations below).
+
+We are using [Playwright](https://playwright.dev/) as the testing framework. This provides browser instances augmented with Playwright's language API for selecting DOM elements, emulating user interactions and making assertions.
+
+## Testing a User's Perspective
+
+A primary benefit of E2E Testing (also Acceptance Testing in this context) is that the tests have zero or very limited visibility of the internals of the system under test. This yields a focus on the user facing feature set irrespective of the code, facilitating refactor without test rewrite and producing a test suite styled around the product requirements and a test output that describes the product's specification.
+
+The tests then should interpret the rendered UI as would a user or an assistive technology. Typically source code would be annotated with HTML attributes for the tests to select on, or tests would make use of DOM XPaths or CSS selector paths. These approaches should be avoided as they incur a reliance on the implementation detail, can be brittle and do not reflect a user's view of the system. Playwright's best practices include a recommendation to [test user-visible behavior](https://playwright.dev/docs/best-practices#test-user-visible-behavior) and an element [Locator](https://playwright.dev/docs/locators) preference to select on accessibility attributes and textual content. Where practical, this approach should be followed.
```

Comment:
Nit: I wouldn't say typically. (My sense is "historically" might be accurate.)

IMO:
> The tests then should interpret the rendered UI as would a user or an assistive technology. Prefer [locating](https://playwright.dev/docs/locators) elements using accessibility attributes or text content and asserting about [their user-visible behavior](https://playwright.dev/docs/best-practices#test-user-visible-behavior). Where necessary, use `data-testid`, css selectors, or XPath. These latter approaches are more fragile since they rely on implementation details.